src/functionalTest/groovy/com/autonomousapps/jvm/IncludedBuildSpec.groovy

```diff
@@ -63,11 +83,73 @@ final class IncludedBuildSpec extends AbstractJvmSpec {
     def project = new IncludedBuildWithSubprojectsProject(true)
     gradleProject = project.gradleProject
 
-    when: 'build does not fail'
+    when:
     build(gradleVersion, gradleProject.rootDir, ':second-build:buildHealth')
 
-    then: 'and there is no advice'
-    assertThat(project.actualIncludedBuildHealth()).containsExactlyElementsIn(project.expectedIncludedBuildHealth)
+    then:
+    assertThat(project.actualIncludedBuildHealth()).containsExactlyElementsIn(project.expectedIncludedBuildHealth('second-build'))
+
+    where:
+    gradleVersion << INCLUDED_BUILD_SUPPORT_GRADLE_VERSIONS
+  }
+
+  def "result of analysis does not change if root build of included build tree changes (with subprojects)"() {
+    given:
+    def project = new IncludedBuildWithSubprojectsProject(true)
+    gradleProject = project.gradleProject
+
+    when:
+    build(gradleVersion, new File(gradleProject.rootDir, 'second-build'), ':buildHealth')
+
+    then:
+    assertThat(project.actualIncludedBuildHealth()).containsExactlyElementsIn(project.expectedIncludedBuildHealth(':'))
+
+    where:
+    gradleVersion << INCLUDED_BUILD_SUPPORT_GRADLE_VERSIONS
+  }
+
+  def "up-to-date check is correct when switching root builds"() {
+    shouldClean = false
```

Comment:
how is this being used?

Reply:
whups. Missed that from debugging. Removed.